domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -55,8 +57,10 @@ public async Task<CreateContractResult> Create(string gsrn, string meteringPoint
                 MeteringPointType = MeteringPointType.Production,
                 MeteringPointOwner = meteringPointOwner,
                 StartDate = startDate,
+                EndDate = DateTimeOffset.Compare(endDate, startDate) >= 1 ? endDate : null,
```

Comment:
This check is setting the EndDate to the passed value of EndDate. Since we're passing a DateTimeOffset, im comparing them to find out whether or not the user has set an EndDate. If the EndDate is bigger than the StartDate. We're setting the EndDate to the passed value.
If the user hasn't set an EndDate, then the default value is passed. Then this will set the EndDate to Null